corehq/sql_db/connections.py

```diff
@@ -36,11 +37,59 @@ def _get_db_alias_or_none(enigne_id):
         return None
 
 
+_thread_locals = local()
+
+
+READ_FROM_CITUS_STANDBYS = 'READ_FROM_CITUS_STANDBYS'
+
+
+def allow_read_from_citus_standbys():
+    return getattr(_thread_locals, READ_FROM_CITUS_STANDBYS, False)
+
+
+class read_from_citus_standbys(ContextDecorator):
+    def __enter__(self):
+        setattr(_thread_locals, READ_FROM_CITUS_STANDBYS, True)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        setattr(_thread_locals, READ_FROM_CITUS_STANDBYS, False)
+
+
+def is_citus_db(connection):
+    """
+    :param connection: either a sqlalchemy connection or a Django cursor
+    """
+    res = connection.execute("SELECT 1 FROM pg_extension WHERE extname = 'citus'")
+    if res is None:
+        res = list(connection)
+    return bool(list(res))
+
+
+def is_citus(engine, citus_list=[]):
+    # hacking mutable default arg to memoize list of citus DBs
+    #   to avoid hitting DB. citus_list is never to be actually
+    #   passed.
```

Comment:
why not just use `memoized`?

Reply:
Does memoize handle the engine object?

Reply:
ah, good point. it probably won't since the engine object may be different between calls. The other issue with this implementation is that it will only 'remember' URLs that are citus.